test/OpCopyMemory.spvasm

```diff
@@ -0,0 +1,41 @@
+; Check SPIRVReader support for OpCopyMemory.
+
+; REQUIRES: spirv-as
+; RUN: spirv-as --target-env spv1.0 -o %t.spv %s
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o - | FileCheck %s
+
+               OpCapability Addresses
+               OpCapability Int16
+               OpCapability Kernel
+               OpMemoryModel Physical64 OpenCL
+               OpEntryPoint Kernel %kernel "copymemory"
+               OpName %pShort "pShort"
+               OpName %dstShort "dstShort"
+               OpName %pInt "pInt"
+               OpName %dstInt "dstInt"
+     %ushort = OpTypeInt 16 0
+       %uint = OpTypeInt 32 0
+       %void = OpTypeVoid
+ %gptr_short = OpTypePointer CrossWorkgroup %ushort
+ %pptr_short = OpTypePointer Function %ushort
+   %gptr_int = OpTypePointer CrossWorkgroup %uint
+   %pptr_int = OpTypePointer Function %uint
+ %kernel_sig = OpTypeFunction %void %gptr_short %gptr_int
+  %ushort_42 = OpConstant %ushort 42
+  %uint_4242 = OpConstant %uint 4242
+     %kernel = OpFunction %void None %kernel_sig
+   %dstShort = OpFunctionParameter %gptr_short
+     %dstInt = OpFunctionParameter %gptr_int
+      %entry = OpLabel
+     %pShort = OpVariable %pptr_short Function %ushort_42
+       %pInt = OpVariable %pptr_int Function %uint_4242
+               OpCopyMemory %dstShort %pShort
+               OpCopyMemory %dstInt %pInt
+               OpReturn
+               OpFunctionEnd
+
+; CHECK-LABEL: define spir_kernel void @copymemory(ptr addrspace(1) %dstShort, ptr addrspace(1) %dstInt)
+; CHECK: call void @llvm.memcpy.p1.p0.i64(ptr addrspace(1) %dstShort, ptr @pShort, i64 ptrtoint (ptr getelementptr (i16, ptr null, i32 1) to i64), i1 false)
+; CHECK: call void @llvm.memcpy.p1.p0.i64(ptr addrspace(1) %dstInt, ptr @pInt, i64 ptrtoint (ptr getelementptr (i32, ptr null, i32 1) to i64), i1 false)
```

Comment:
I'm open to suggestions!  This is what `llvm::ConstantExpr::getSizeOf` generates, which should work for any type (including structs).  That should be more robust than an ad-hoc type size calculation, but I'm happy to consider alternatives if they are robust enough.

Reply:
Speaking very-very strictly, I believe that `getSizeOf` is not a perfect match in this case. `llvm.memcpy.*` expects `len` to be in bytes, while `getSizeOf` returns a size in a number of 'address units' which is 1 byte in ARM and X86 architectures (and guess in all of GPUs), but it might not be the case for various DSPs, so there such `memcpy` would be broken... Or is address size is stored in the target's datalayout? If so - no objections form me.

Reply:
Perhaps we can just call `DataLayout::getTypeSizeInBits()`; it seems to handle aggregate types too and it will return a single number.  I'll try and update the patch.

Reply:
I'm fine with this as-is, but we might want to consider `DataLayout::getTypeStoreSize()` instead.  I don't think this will make a difference in practice right now, but `getTypeStoreSize()` returns the number of bytes directly, and should handle any weird types in the future where the number of bits is not a multiple of 8.

Reply:
Sounds reasonable to make it more future-proof.  Updated.